File: 05C++/ex1_3.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

int calculateLcd(int dividend, int divisor)
{
    int remainder = dividend % divisor;

    if (remainder == 0) return divisor;
    calculateLcd(divisor, remainder);
}

int main(int argc, char *argv[])
{
    int dividend = atoi(argv[1]);
    int divisor = atoi(argv[2]);

    int result = calculateLcd(dividend, divisor);

    std::cout << result << endl;
    return 0;
}

File: 08Ruby/ex1_3.rb
class App
  def calculate_lcd(dividend, divisor)
    remainder = dividend % divisor
    if remainder == 0 then return divisor end
    return calculate_lcd divisor, remainder
  end

  def main
    dividend = ARGV[0].to_i
    divisor = ARGV[1].to_i

    result = calculate_lcd(dividend, divisor)

    puts "#{result}"
  end
end

if __FILE__ == $0
  app = App.new
  app.main
end
